src/protocols/minecraft/types.rs

```diff
@@ -238,3 +252,10 @@ pub(crate) fn get_string<B: ByteOrder>(buffer: &mut Buffer<B>) -> GDResult<Strin
 
     String::from_utf8(text).map_err(|e| PacketBad.context(e))
 }
+
+pub(crate) fn as_string(value: &str) -> Vec<u8> {
+    let mut buf = as_varint(value.len() as i32);
```

Comment:
I'm nitpicking but this would surely wrap to negative if greater than `i32::MAX` (very unlikely), we could use `try_into()` to catch that instead of sending a potentially malformed packet in that case (I could be misunderstanding `as_varint`). Altough to be honest i'm sure the same conversion is done elsewhere.

Reply:
I wouldn't ever think that we'll have a String passed here greater than `2_147_483_647`, but this could be added for safety.